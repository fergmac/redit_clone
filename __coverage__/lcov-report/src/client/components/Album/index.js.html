<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/components/Album/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/client/components/Album</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PropTypes } from 'react';
// import styles from './style.css';
// import Chip from 'material-ui/Chip';
import Card from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
// import { Link } from 'react-router';
const styles = {
  albumContainer: {
    /* display:flex;
    flex-flow: column nowrap;
    justify-content: space-between;*/
    padding: '1rem',
    marginTop: '1rem',
  },
  a: {
    textDecoration: 'none',
  },
};
// was post
const Album = ({ album, voteUp }) =&gt; (
  &lt;Card className={styles.albumContainer}&gt;
    {/* &lt;Link to={album.link}&gt;&lt;h3&gt;{album.title}&lt;/h3&gt;&lt;/Link&gt; */}
    &lt;a href="'`${album.link}`'"&gt;&lt;h3&gt;{album.title}&lt;/h3&gt;&lt;/a&gt;
    &lt;p&gt;{album.description}&lt;/p&gt;
      {/* {post.lessons.map((lesson, index) =&gt; (&lt;Chip key={index}&gt;{lesson}&lt;/Chip&gt;))} */}
    &lt;FlatButton
      label={`${album.votes} Votes`}
      onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >voteUp(album.id)}</span>
      primary
    /&gt;
  &lt;/Card&gt;
);
&nbsp;
Album.propTypes = {
  album: PropTypes.object.isRequired,
  voteUp: PropTypes.object.isRequired,
};
&nbsp;
export default Album;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Nov 09 2016 16:25:23 GMT-0800 (PST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
