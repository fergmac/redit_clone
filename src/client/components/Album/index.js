import React, { PropTypes } from 'react';
import styles from './style.css';
// import Chip from 'material-ui/Chip';
import Card from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
// import { Link } from 'react-router';

// was post
const Album = ({ album, voteUp }) => (
  <Card className={styles.albumContainer}>
    {/* <Link to={album.link}><h3>{album.title}</h3></Link> */}
    <a href="'`${album.link}`'"><h3>{album.title}</h3></a>
    <p>{album.description}</p>
      {/* {post.lessons.map((lesson, index) => (<Chip key={index}>{lesson}</Chip>))} */}
    <FlatButton label={`${album.votes} Votes`} primary onClick={voteUp.bind(this, album.id)} />
  </Card>
);

Album.propTypes = {
  album: PropTypes.object.isRequired,
  voteUp: PropTypes.func.isRequired,
};

export default Album;
