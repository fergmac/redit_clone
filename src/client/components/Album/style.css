.albumContainer {
    /*display:flex;
    flex-flow: column nowrap;
    justify-content: space-between;*/
    padding: 1rem;
    margin-top: 1rem;
}

a {
    text-decoration: none;
}
