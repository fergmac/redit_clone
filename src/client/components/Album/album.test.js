/* eslint-disable */
import React, { PropTypes } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Album from './index';


describe('Album', () => {
    const albumData = {
        id: 1,
        title: 'Jazz Guitar',
        description: '1920s acoustic guitar jazz',
        votes: [{ id: 1 }]
    }
    const voteUps = {
        
    }
    it('renders the albums', () => {
       const wrapper = shallow(
                <Album 
                    album={albumData}
                    voteUp={voteUps}
                />
       )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})
