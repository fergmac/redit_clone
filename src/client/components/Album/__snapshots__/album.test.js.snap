exports[`Album renders the albums 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
  style={
    Object {
      "marginTop": "1rem",
      "padding": "1rem",
    }
  }>
  <a
    href="\'\`\${album.link}\`\'">
    <h3>
      Jazz Guitar
    </h3>
  </a>
  <p>
    1920s acoustic guitar jazz
  </p>
  <FlatButton
    disabled={false}
    label="[object Object] Votes"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={true}
    secondary={false} />
</Card>
`;
